common/src/common.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::HashSet;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Display};
+use tokio::sync::mpsc::UnboundedSender;
+use uuid::Uuid;
+
+type WSSender = UnboundedSender<Result<warp::ws::Message, warp::Error>>;
+
+#[derive(Serialize, Deserialize)]
+pub enum ReqData {
+    HeartbeatData(Uuid),
+    CreateRoomData(String),
+    GetRoomData(String),
+    JoinRoomData(String, Uuid, Uuid),
+    SendMsgData(ChatMessage, Uuid),
+    LoginData(String),
+    RegistrationData(String),
+    LeaveRoomData(Uuid, Uuid),
+    ExitAppData(Uuid),
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ChatMessage {
+    pub author: String,
+    pub contents: String,
+    pub timestamp: DateTime<Utc>,
+}
+
+impl Display for ChatMessage {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "[{}] {}: {}", self.timestamp, self.author, self.contents)
+    }
+}
+
+impl ChatMessage {
+    pub fn new(author: &str, contents: &str) -> ChatMessage {
+        ChatMessage {
+            author: author.to_string(),
+            contents: contents.to_string(),
+            timestamp: Utc::now(),
```

Comment:
Bardziej praktyczne jest użycie `Local` zamiast `Utc`. Natomiast jeśli byśmy zakładali, że nasi użytkownicy mogą się znajdować w różnych strefach czasowych to wtedy powinniśmy przeprowadzić konwersję przed wypisaniem daty.